packages/flutter/lib/src/material/slider.dart

```diff
@@ -907,152 +918,113 @@ class _RenderSlider extends RenderBox {
-    final ColorTween inactiveTickMarkEnableColor = ColorTween(begin: _sliderTheme.disabledInactiveTickMarkColor, end: _sliderTheme.inactiveTickMarkColor);
-
-    final Paint activeTrackPaint = Paint()..color = activeTrackEnableColor.evaluate(_enableAnimation);
-    final Paint inactiveTrackPaint = Paint()..color = inactiveTrackEnableColor.evaluate(_enableAnimation);
-    final Paint activeTickMarkPaint = Paint()..color = activeTickMarkEnableColor.evaluate(_enableAnimation);
-    final Paint inactiveTickMarkPaint = Paint()..color = inactiveTickMarkEnableColor.evaluate(_enableAnimation);
 
+    // The visual position is the position of the thumb from 0 to 1 from left
+    // to right. In left to right, this is the same as the value, but it is
+    // inverted for right to left text.
     double visualPosition;
-    Paint leftTrackPaint;
-    Paint rightTrackPaint;
-    Paint leftTickMarkPaint;
-    Paint rightTickMarkPaint;
+    bool leftTrackActive;
+    bool rightTrackActive;
+
     switch (textDirection) {
       case TextDirection.rtl:
         visualPosition = 1.0 - value;
-        leftTrackPaint = inactiveTrackPaint;
-        rightTrackPaint = activeTrackPaint;
-        leftTickMarkPaint = inactiveTickMarkPaint;
-        rightTickMarkPaint = activeTickMarkPaint;
+        leftTrackActive = false;
+        rightTrackActive = true;
         break;
       case TextDirection.ltr:
         visualPosition = value;
-        leftTrackPaint = activeTrackPaint;
-        rightTrackPaint = inactiveTrackPaint;
-        leftTickMarkPaint = activeTickMarkPaint;
-        rightTickMarkPaint = inactiveTickMarkPaint;
+        leftTrackActive = true;
+        rightTrackActive = false;
         break;
     }
 
-    const double trackRadius = _trackHeight / 2.0;
-    const double thumbGap = 2.0;
+    final List<bool> trackSegmentsActive = <bool>[leftTrackActive, rightTrackActive];
+    final List<double> trackDividers = <double>[visualPosition];
 
-    final double trackVerticalCenter = offset.dy + (size.height) / 2.0;
-    final double trackLeft = offset.dx + _overlayRadius;
-    final double trackTop = trackVerticalCenter - trackRadius;
-    final double trackBottom = trackVerticalCenter + trackRadius;
-    final double trackRight = trackLeft + trackLength;
-    final double trackActive = trackLeft + trackLength * visualPosition;
-    final double thumbRadius = _sliderTheme.thumbShape.getPreferredSize(isInteractive, isDiscrete).width / 2.0;
-    final double trackActiveLeft = math.max(0.0, trackActive - thumbRadius - thumbGap * (1.0 - _enableAnimation.value));
-    final double trackActiveRight = math.min(trackActive + thumbRadius + thumbGap * (1.0 - _enableAnimation.value), trackRight);
-    final Rect trackLeftRect = Rect.fromLTRB(trackLeft, trackTop, trackActiveLeft, trackBottom);
-    final Rect trackRightRect = Rect.fromLTRB(trackActiveRight, trackTop, trackRight, trackBottom);
+    // Paint the track.
+    _sliderTheme.trackShape.paint(context, offset,
```

Comment:
Style nit:

I think this should have newlines for `context` and `offset`, and have a trailing comma according to [flutter style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation).

Like this:

```
_sliderTheme.trackShape.paint(
  context, 
  offset,
  parentBox: this,
  sliderTheme: _sliderTheme,
  enableAnimation: _enableAnimation,
  trackDividers: trackDividers,
  trackSegmentsActive: trackSegmentsActive,
  isDiscrete: isDiscrete,
  isEnabled: isInteractive,
);
```